tcms/testcases/migrations/0011_trim_bugsystem_fields.py

```diff
@@ -10,6 +41,21 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
+        migrations.RunPython(forwards_store_data, backwards_restore_data),
+        migrations.AlterField(model_name='bugsystem',
+                              name='validate_reg_exp',
+                              preserve_default=False,
+                              field=models.CharField(max_length=128,
+                                                     help_text='A valid JavaScript regular '
+                                                               'expression such as ^\\d$',
+                                                     verbose_name='RegExp for ID validation')),
+        migrations.AlterField(model_name='bugsystem',
+                              name='url_reg_exp',
+                              preserve_default=False,
+                              field=models.CharField(max_length=8192,
+                                                     help_text='A valid Python format string such '
+                                                               'as http://bugs.example.com/%s',
+                                                     verbose_name='URL format string')),
```

Comment:
If these 2 are ever needed IMO they should be at the end of the operations list so that they will be executed first in the reverse order.

Reply:
Already tried, migrations failed since the fields have been removed.

Reply:
> Already tried

What exactly? Be more specific.

Also I see you have a `default=''` which I'm not sure if will work or not.

In any case, if these AlterField operations are to work, they should be at the end of the operations list, not in the middle like you have them now.

Reply:
> What exactly? Be more specific.

I was referring to your point that the Alterfields should be at the end of the operations list. I had already tried it and the migrations failed since the fields have been removed. I've moved them to the end of the operations list and the error is back.

`django.core.exceptions.FieldDoesNotExist: BugSystem has no field named 'validate_reg_exp'`

Reply:
hmm, I need to think how to solve this. The problem is two-fold:

- when you add AlterField at the end of the operations list it will fail when migration is applied forwards b/c the field doesn't exist already, however we need it in this location in order for this to be the 1st operation that is applied in the reverse order.
- when AlterField is in a different place in the operations list that would cause a mismatch between the state of DB and the state of the model wrt the default values of these fields (if preserve_default=False isn't used). However the forward migration is not what we're interested in and having AlterField before RemoveField doesn't make much sense.